physicsnemo/experimental/models/dit/dit.py

```diff
@@ -0,0 +1,354 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Union, Optional, Any
+import torch
+import torch.nn as nn
+import numpy as np
+import math
+
+from timm.models.vision_transformer import Attention
+
+try:
+    from transformer_engine.pytorch import MultiheadAttention
+except ImportError:
+    print("Transformer Engine not found, use 'timm' Attention.")
```

Comment:
If transformer_engine is not installed, but the uses requests it, it will have an unexpected error below.  Instead of quitely changing, I recommend failing loudly in that situation.

See, for example, here:
https://github.com/NVIDIA/physicsnemo/blob/main/physicsnemo/models/transolver/transolver.py#L129-L132

Could you remove the printouts and instead raise an exception if the required module is requested in the constructor but not available?